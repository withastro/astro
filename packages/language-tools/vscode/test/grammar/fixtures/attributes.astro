<!-- Test special characters inside attributes names -->

<div @class></div>
<div class_name></div>
<div class.astro></div>
<div i18nReady></div>

<div onclick="console.log()"></div>

<div onload={"console.log('')"}></div>

<div onload=`console.log('')`></div>

<div class=`\${not-a-variable}`></div>
