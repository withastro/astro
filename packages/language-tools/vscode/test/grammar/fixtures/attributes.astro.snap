><!-- Test special characters inside attributes names -->
#^^^^ source.astro comment.block.astro punctuation.definition.comment.astro
#    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.astro comment.block.astro
#                                                     ^^^ source.astro comment.block.astro punctuation.definition.comment.astro
>
><div @class></div>
#^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.begin.astro
# ^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro entity.name.tag.astro
#    ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#     ^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.@class.astro entity.other.attribute-name.astro
#           ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.end.astro
#            ^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.begin.astro
#              ^^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro entity.name.tag.astro
#                 ^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.end.astro
><div class_name></div>
#^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.begin.astro
# ^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro entity.name.tag.astro
#    ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#     ^^^^^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.class_name.astro entity.other.attribute-name.astro
#               ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.end.astro
#                ^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.begin.astro
#                  ^^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro entity.name.tag.astro
#                     ^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.end.astro
><div class.astro></div>
#^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.begin.astro
# ^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro entity.name.tag.astro
#    ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#     ^^^^^^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.class.astro.astro entity.other.attribute-name.astro
#                ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.end.astro
#                 ^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.begin.astro
#                   ^^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro entity.name.tag.astro
#                      ^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.end.astro
><div i18nReady></div>
#^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.begin.astro
# ^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro entity.name.tag.astro
#    ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#     ^^^^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.i18nReady.astro entity.other.attribute-name.astro
#              ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.end.astro
#               ^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.begin.astro
#                 ^^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro entity.name.tag.astro
#                    ^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.end.astro
>
><div onclick="console.log()"></div>
#^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.begin.astro
# ^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro entity.name.tag.astro
#    ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#     ^^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onclick.astro entity.other.attribute-name.astro
#            ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onclick.astro punctuation.separator.key-value.astro
#             ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onclick.astro meta.embedded.line.js string.quoted.astro punctuation.definition.string.begin.astro
#              ^^^^^^^^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onclick.astro meta.embedded.line.js string.quoted.astro
#                           ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onclick.astro meta.embedded.line.js string.quoted.astro punctuation.definition.string.end.astro
#                            ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.end.astro
#                             ^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.begin.astro
#                               ^^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro entity.name.tag.astro
#                                  ^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.end.astro
>
><div onload={"console.log('')"}></div>
#^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.begin.astro
# ^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro entity.name.tag.astro
#    ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#     ^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onload.astro entity.other.attribute-name.astro
#           ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onload.astro punctuation.separator.key-value.astro
#            ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onload.astro punctuation.section.embedded.begin.astro
#             ^^^^^^^^^^^^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onload.astro meta.embedded.expression.astro source.tsx
#                              ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onload.astro punctuation.section.embedded.end.astro
#                               ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.end.astro
#                                ^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.begin.astro
#                                  ^^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro entity.name.tag.astro
#                                     ^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.end.astro
>
><div onload=`console.log('')`></div>
#^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.begin.astro
# ^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro entity.name.tag.astro
#    ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#     ^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onload.astro entity.other.attribute-name.astro
#           ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onload.astro punctuation.separator.key-value.astro
#            ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.onload.astro
#             ^^^^^^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.console.log.astro entity.other.attribute-name.astro
#                        ^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#                             ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.end.astro
#                              ^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.begin.astro
#                                ^^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro entity.name.tag.astro
#                                   ^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.end.astro
>
><div class=`\${not-a-variable}`></div>
#^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.begin.astro
# ^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro entity.name.tag.astro
#    ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#     ^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.class.astro entity.other.attribute-name.astro
#          ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.class.astro punctuation.separator.key-value.astro
#           ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.class.astro
#            ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#             ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.attribute.$.astro entity.other.attribute-name.astro
#              ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#               ^^^^^^^^^^^^^^ source.astro meta.scope.tag.div.astro meta.tag.start.astro meta.embedded.expression.astro source.tsx
#                             ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#                              ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro
#                               ^ source.astro meta.scope.tag.div.astro meta.tag.start.astro punctuation.definition.tag.end.astro
#                                ^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.begin.astro
#                                  ^^^ source.astro meta.scope.tag.div.astro meta.tag.end.astro entity.name.tag.astro
#                                     ^ source.astro meta.scope.tag.div.astro meta.tag.end.astro punctuation.definition.tag.end.astro
>